docs/articles/contributing/architecture.md

```diff
@@ -1,5 +1,40 @@
 
-All the references to timely from other parts of the docs might instead point to a subsection here
+## Bytewax from 10,000 feet
 
-Bytewax is a Python native binding to the [Timely Dataflow](https://github.com/TimelyDataflow/timely-dataflow) library. Timely Dataflow is a dataflow processing library written in [Rust](https://www.rust-lang.org/).
+### Timely Roots
+
+Bytewax has its roots in the Rust project [Timely Dataflow]() and although it is not necessary to understand Timely Dataflow to use Bytewax, it is a core component (the distributed processing engine) of Bytewax and many of the concepts are passed through. The short history of Timely Dataflow is that there were a few folks ([Frank McSherry] and [Derek G. Murray]) working for Microsoft Research about a decade ago that worked on a project called [Naiad](https://www.youtube.com/watch?v=yyhMI9r0A9E). Naiad was a system designed around the concept of how you could create a distributed system to execute dataflow programs that would be performant across batch, stream, and graph processing. From the [paper abstract](https://dl.acm.org/doi/10.1145/2517349.2522738): 
+> Naiad is a distributed system for executing data parallel, cyclic dataflow programs. It offers the high throughput of batch processors, the low latency of stream processors, and the ability to perform iterative and incremental computations. Although existing systems offer some of these features, applications that require all three have relied on multiple platforms, at the expense of efficiency, maintainability, and simplicity. Naiad resolves the complexities of combining these features in one framework.
+> A new computational model, timely dataflow, underlies Naiad and captures opportunities for parallelism across a wide class of algorithms. This model enriches dataflow computation with timestamps that represent logical points in the computation and provide the basis for an efficient, lightweight coordination mechanism.
+> We show that many powerful high-level programming models can be built on Naiad's low-level primitives, enabling such diverse tasks as streaming data analysis, iterative machine learning, and interactive graph mining. Naiad outperforms specialized systems in their target application domains, and its unique features enable the development of new high-performance applications.
+
+At some point, Microsoft Research changed direction with respect to the Silicon Valley Lab (I don't know the real story, but some of the prior reporting on it is [here](https://www.vox.com/2014/9/18/11631044/microsoft-shuts-down-silicon-valley-research-lab-amid-broader-layoffs)) and in some way or another, the team moved on from there to start other things. Many of the original team kept the idea of dataflow programs core to what they next worked on. Frank went on to spend time working on a Rust version of the core tenants of Naiad and Derek went on to bring the core dataflow programming concepts to tensorflow.
```

Comment:
I suggest changing the to passive voice in parens
_(the real story is unclear, but some of the prior reporting on it is...)_